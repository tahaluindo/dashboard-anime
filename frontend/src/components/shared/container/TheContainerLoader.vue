<template>
  <div ref="element">
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent, onMounted, ref } from "vue";
import { Observer } from "../../../controllers/Observer";
import { getElement, injectStrict } from "../../../helpers/utils";
import { elementObserverKey } from "./controller/elementObserver";

export default defineComponent({
  setup() {
    const observer = injectStrict<Observer>(elementObserverKey);
    const element = ref<HTMLElement>();

    onMounted(() => {
      const elementValue = getElement(element.value);
      observer.observe(elementValue);
    });

    return {
      element,
    };
  },
});
</script>

<style lang="postcss"></style>
