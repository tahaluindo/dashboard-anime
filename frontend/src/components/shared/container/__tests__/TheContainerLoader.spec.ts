import { mount } from "@vue/test-utils";
import { mockWindowIntersectionObserver } from "../../../../__mocks__/window.mock";
import {
  ElementObserver,
  elementObserverKey,
} from "../controller/elementObserver";
import TheContainerLoader from "../TheContainerLoader.vue";

describe("TheContainerLoader component test", () => {
  it("should render the default component", async () => {
    mockWindowIntersectionObserver();
    const elementObserver = new ElementObserver();

    const wrapper = mount(TheContainerLoader, {
      shallow: true,
      global: {
        provide: {
          [elementObserverKey as symbol]: elementObserver,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
