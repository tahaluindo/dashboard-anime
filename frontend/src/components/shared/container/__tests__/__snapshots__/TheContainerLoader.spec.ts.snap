// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheContainerLoader component test should render the default component 1`] = `"<div></div>"`;
